refiner-lib/src/refiner_inner.rs

```diff
@@ -395,18 +402,18 @@ fn build_transaction(
                 }
                 InnerTransactionKind::Call => {
                     hash = virtual_receipt_id.0.try_into().unwrap();
-                    tx = tx.hash(hash).from(near_account_to_evm_address(
-                        outcome.receipt.predecessor_id.as_bytes(),
-                    ));
+                    let from_address =
```

Comment:
I'm not actually using the from_address to create the legacy address, since it is just 
`contract_address(None)` on line 425. Is it Ok?

Reply:
Yes, this is good. The `call` method must always have a target contract so it cannot be a top-level contract creation. 

This doesn't catch the case where the contract called is a factory, so a new contract is deployed by an internal transaction, but this is ok because Ethereum doesn't catch that either (for example the `contract_address` field of the transaction `0xeba52a52e94894315bb58aad405252e4307ad413da0c505d663d9d2ad5e3c19e` is `null` even though it created the contract at `0x415F48A2Bb102270712e28026A5A4557d4696B0f` with an internal transaction).